File: atcoder/abc215_d_Prime_factorization.cpp
// https://atcoder.jp/contests/abc215/tasks/abc215_d
#include<iostream>
// #include<algorithm>
// #include<string>
// #include<numeric>
// #include<vector>
// #include<map>
// #include<tuple>
// #include<set>
// #include<queue>
// #include<deque>
// #include<regex>
// #include<bitset>
// #include<iomanip>
// #include<complex>
// #include<stack>
// #include<functional>

#include<bits/stdc++.h>
using namespace std;
// #include <atcoder/all>
// using namespace atcoder;
#define rep(i,n) for (int i = 0; i < (n); ++i)
#define drep(i,n) for(int i = (n-1); i >= 0; i--)
#define all(v) (v).begin(),(v).end()
#define maxs(x,y) (x = max(x,y))
#define mins(x,y) (x = min(x,y))
template <class T> bool chmax(T &a, const T &b) { if (a < b) { a = b; return 1; } return 0; }
template <class T> bool chmin(T &a, const T &b) { if (b < a) { a = b; return 1; } return 0; }
template <class T> T gcd(T a, T b) { return b ? gcd(b, a % b) : a; }
template <class T> T lcm(T a, T b) { return a/gcd(a,b)*b; }
typedef pair<int, int> P;
typedef long long ll;
const int INF = 1001001001;
const ll LINF = 1001002003004005006ll;
const ll MOD = 1e9+7;

int main() {
    int n, m;
    cin >> n >> m;
    const int L = 100001;
    vector<bool> x(L);
    rep(i,n) {
        int a;
        cin >> a;
        x[a] = true;
    }

    vector<int> d; // Prime factorization
    for (int i = 2; i < L; i++) {
        bool flag = false;
        for (int j = i; j < L; j += i) {
            if (x[j]) flag = true; // if a%i==0
        }
        if (flag) d.push_back(i);
    }

    vector<bool> ok(m+1, true);
    for (int i :d) {
        for (int j = i; j <= m; j += i){
            ok[j] = false;
        }
    }
    int cnt = 0;
    for (int i = 1; i <= m; ++i) if (ok[i]) cnt++;
    cout << cnt << endl;
    for (int i = 1; i <= m; ++i) if (ok[i]) cout << i << endl;
	return 0;
}

File: atcoder/abc215_d_Prime_factorization.py
# https://atcoder.jp/contests/abc215/tasks/acc215_d

N, M = map(int, input().split())
A = list(map(int, (input().split())))
L = 100001
x = [False] * L
for i in range(N):
    x[A[i]] = True

d = []
for i in range(2, L):
    flag = False
    for j in range(i, L, i):
        if x[j]:
            flag = True
            break
    if flag: d.append(i)

ok = [True] * (M+1)
for i in d:
    for j in range(i, M+1, i):
        ok[j] = False

cnt = 0
for i in range(1, M+1):
    if ok[i]:
        cnt += 1
print(cnt)
for i in range(1, M+1):
    if ok[i]:
        print(i)


# import math
# import collections
# def prime_factor_table(n):
#     table = [0] * (n + 1)
    
#     for i in range(2, n + 1):
#         if table[i] == 0:
#             for j in range(i + i, n + 1, i):
#                 table[j] = i
    
#     return table

# def prime_factor(n, prime_factor_table):
#     prime_count = collections.Counter()
    
#     while prime_factor_table[n] != 0:
#         prime_count[prime_factor_table[n]] += 1
#         n //= prime_factor_table[n]
#     prime_count[n] += 1
    
#     return prime_count

# N, M = map(int, input().split())
# A = list(map(int, (input().split())))
# MAXA = M
# for a in A:
#     if a>MAXA: MAXA = a
# table = prime_factor_table(MAXA)

# factorization = set()
# for a in A:
#     if a<2: continue
#     factorization |= set(prime_factor(a, table))
# # print(factorization)
# ans = [1]
# for k in range(2, M+1):
#     kfact = set(prime_factor(k, table))
#     # print(k, kfact, factorization)
#     if kfact.isdisjoint(factorization):
#         ans.append(k)
# print(len(ans))
# for k in ans:
#     print(k)
